presto-thrift-interface/src/main/java/com/facebook/presto/thrift/interfaces/client/ThriftSingleValue.java

```diff
@@ -0,0 +1,88 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift.interfaces.client;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.predicate.Marker;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.thrift.interfaces.readers.ColumnReaders;
+import com.facebook.presto.thrift.interfaces.writers.ColumnWriter;
+import com.facebook.presto.thrift.interfaces.writers.ColumnWriters;
+import com.facebook.swift.codec.ThriftConstructor;
+import com.facebook.swift.codec.ThriftField;
+import com.facebook.swift.codec.ThriftStruct;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.thrift.interfaces.client.ThriftRangeValueSet.ThriftRange.Bound.toMarkerBound;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+@ThriftStruct
+public final class ThriftSingleValue
+{
+    public static final String VALUE_COLUMN_NAME = "value";
+    private final List<ThriftColumnData> columnsData;
+
+    @ThriftConstructor
+    public ThriftSingleValue(List<ThriftColumnData> columnsData)
+    {
+        this.columnsData = requireNonNull(columnsData, "columnsData is null");
+        checkArgument(!columnsData.isEmpty(), "Columns data is empty");
+    }
+
+    @ThriftField(1)
+    public List<ThriftColumnData> getColumnsData()
+    {
+        return columnsData;
+    }
+
+    @Nullable
+    public static ThriftSingleValue fromMarker(Marker marker)
+    {
+        if (!marker.getValueBlock().isPresent()) {
+            return null;
+        }
+        ColumnWriter writer = ColumnWriters.create(VALUE_COLUMN_NAME, marker.getType(), 1);
+        Block valueBlock = marker.getValueBlock().get();
+        checkState(valueBlock.getPositionCount() == 1,
+                "Block in Marker has more than one position: %s", valueBlock.getPositionCount());
+        writer.append(valueBlock, 0, marker.getType());
+        return new ThriftSingleValue(writer.getResult());
+    }
+
+    public static Marker toMarker(@Nullable ThriftSingleValue value, Type type, ThriftRangeValueSet.ThriftRange.Bound bound)
+    {
+        if (value == null) {
+            switch (bound) {
+                case ABOVE:
+                    return Marker.lowerUnbounded(type);
+                case BELOW:
+                    return Marker.upperUnbounded(type);
+                case EXACTLY:
+                    throw new IllegalArgumentException("Value cannot be null for 'EXACTLY' bound");
+                default:
+                    throw new IllegalArgumentException("Unknown bound type: " + bound);
+            }
+        }
+        else {
+            Block block = ColumnReaders.readBlock(value.getColumnsData(), VALUE_COLUMN_NAME, type, 1);
```

Comment:
I'm not sure if it's a good idea to statically import this method. First of all, it's not a well-known library method. Next, all reader classes have a method with the same name, so including class name makes it clear which one is called.
However, I don't have a strong preference about it, so if there are good reasons why it should be statically imported then I can do it.